File: 프로그래머스/unrated/181921. 배열 만들기 2/배열 만들기 2.py
def solution(l, r):

    result = []

    for i in range(l, r+1):
        # 5로 나누어 떨어지지 않으면 건너뜀
        if i % 5 != 0:
            continue

        # 숫자자리마다 0 또는 5인지 판별하기 위한 변수 ch
        ch = str(i) 
        # 0또는 5만 있으면 True
        is_zero_or_five = True

        # 숫자 전체를 돎
        for j in ch:
            # 만약 각 자리가 5또는 0이 아니라면
            if j != "5" and j != "0":            
                is_zero_or_five = False

        # True로 0,5만 있는 숫자라면 result에 더함
        if is_zero_or_five:
            result.append(i)

    # print(result)

    if not result:
        return [-1]
    return result

File: 프로그래머스/unrated/181921. 배열 만들기 2/배열 만들기 2.java
import java.util.List;
import java.util.Arrays;
import java.util.ArrayList;

class Solution {
    public List<Integer> solution(int l, int r) {
        List<Integer> result = new ArrayList<Integer>();
        
        for(int num = l; num <= r; num++) {        
            // 5로 나누어 떨어지지 않으면 넘김
            if(num%5 != 0){
                continue;
            }
            
            String num_word = String.valueOf(num);
            boolean five_and_zero = true;
            
            // 문자를 순회
            for(int i = 0; i < num_word.length(); i++) {
                char chr = num_word.charAt(i);
                // 문자 중 하나라도 0, 5가 아니면 false
                if(chr != '0' && chr != '5') {
                    five_and_zero = false;
                }
            }
            
            // 0, 5가 아닌 값이 없으면 결과
            if(five_and_zero) {
                result.add(Integer.valueOf(num_word));
            }
        }
        
        // 결과가 없으면 -1
        if(result.size() == 0){
            return Arrays.asList(-1);
        }
        
        return result;
    }
}